applications/StructuralMechanicsApplication/python_scripts/adjoint_structural_solver.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+# importing the Kratos Library
+import KratosMultiphysics
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+import structural_mechanics_solver
+
+# Check that KratosMultiphysics was imported in the main script
+KratosMultiphysics.CheckForPreviousImport()
+
+def CreateSolver(main_model_part, custom_settings):
+    return AdjointStructuralSolver(main_model_part, custom_settings)
+
+class AdjointStructuralSolver(structural_mechanics_solver.MechanicalSolver):
```

Comment:
`StructuralMechanicsAdjointSolver`

Furthermore, this works only for static problems right?
Then it should be e.g. `StructuralMechanicsAdjointStaticSolver`